src/Content/Lock.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace Kirby\Content;
+
+use Kirby\Cms\App;
+use Kirby\Cms\User;
+use Kirby\Toolkit\Str;
+
+/**
+ * The Lock class provides information about the
+ * locking state of a content version, depending
+ * on the timestamp and locked user id
+ *
+ * @internal
+ * @since 5.0.0
+ *
+ * @package   Kirby Content
+ * @author    Bastian Allgeier <bastian@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ */
+class Lock
+{
+	public function __construct(
+		protected User|null $user = null,
+		protected int|null $modified = null,
+	) {
+	}
+
+	/**
+	 * Creates a lock for the given version by
+	 * reading the modification timestamp and
+	 * lock user id from the version.
+	 */
+	public static function for(
+		Version $version
+	): static {
+		// if the version does not exist, it cannot be locked
+		if ($version->exists() === false) {
+			// create an open lock for the current user
+			return new static(
+				user: App::instance()->user(),
+				modified: null
+			);
+		}
+
+		// Read the locked user id from the version
+		$userId = $version->read('default')['lock'] ?? null;
+
+		// No user? Create the lock for the currently authenticated user
+		$user = App::instance()->user($userId) ?? App::instance()->user();
```

Comment:
`Question:` I don't quite understand why we are not passing null in this case but the current user. Shouldn't there be a difference between "lock is set but user not found/deleted" and "no lock is set"?